src/libstd/sync/rwlock.rs

```diff
@@ -64,8 +64,8 @@ pub struct RwLock<T> {
     data: UnsafeCell<T>,
 }
 
-unsafe impl<T:'static+Send> Send for RwLock<T> {}
-unsafe impl<T> Sync for RwLock<T> {}
```

Comment:
I was also surprised that `RwLock` previously had such loose bounds, however, I think it was impossible to construct one that didn't satisfy them, so it has probably not mattered (fortunately).